lib/scrabble.rb

```diff
@@ -0,0 +1,64 @@
+# require_relative '../lib/scrabble_player'
+
+module Scrabble
+
+  class Scoring
+
+    attr_reader :SCOREBOARD
+
+    # constant that is a hash of the point values of the letters
+    SCOREBOARD = {
+      %w[a e i o u l n r s t]=>1,
+      %w[d g]=> 2, %w[b c m p]=>3,
+      %w[f h v w y]=> 4,
+      %w[k]=> 5,
+      %w[j x]=> 8,
+      %w[q z]=> 10
+    }
+
+    def self.score word
+
+      letter_values = []
+
+      raise ArgumentError.new "Error. Please enter a string." if word.class != String
+
+      raise ArgumentError.new "Error. Please enter a string of letters." if word.match(/^[[:alpha:]]+$/) == nil
+
+      raise ArgumentError.new "Error. Please enter a string that is less than 7 characters." if word.length > 7
+
+      word_array = word.downcase.scan /\w/ # regex that checks to make sure the string consists of only letters
+
+      # this checks the value of our letters
+      SCOREBOARD.each do |key, value|
+        word_array.each do |letter|
+          key.include?(letter) ? letter_values << value : letter_values << 0
+        end
+      end
```

Comment:
This seems a little convoluted with nested `each` loops.  You could either simplify it using the hash I outlined above and `map` or use a `select` in the loop you have above to select all the letters inside `word_array` that appear in key and multiply the number by the value.